homeassistant/components/wireguard/__init__.py

```diff
@@ -0,0 +1,32 @@
+"""The WireGuard integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import WireGuardUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WireGuard from a config entry."""
+
+    coordinator: WireGuardUpdateCoordinator = WireGuardUpdateCoordinator(hass, entry)
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
```suggestion
    coordinator = WireGuardUpdateCoordinator(hass, entry)
    await coordinator.async_config_entry_first_refresh()
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```
This is the preferred sequence, so that `hass.data` is only filled with the object when the first refresh was actually successful.